qt/Session.cc

```diff
@@ -83,28 +83,37 @@ void Session::sessionSet(tr_quark const key, QVariant const& value)
 
 void Session::portTest()
 {
-    auto* q = new RpcQueue{};
+    auto const response_func = [this](RpcResponse const& r)
+    {
+        bool is_open = false;
+        std::string_view ip_protocol;
 
-    q->add([this]() { return exec("port-test", nullptr); });
+        if (auto const value = dictFind<std::string_view>(r.args.get(), TR_KEY_ipProtocol); value)
+        {
+            ip_protocol = *value;
+        }
 
-    q->add(
-        [this](RpcResponse const& r)
+        if (auto const value = dictFind<bool>(r.args.get(), TR_KEY_port_is_open); value)
         {
-            bool is_open = false;
+            is_open = *value;
+        }
 
-            if (r.success)
-            {
-                auto const value = dictFind<bool>(r.args.get(), TR_KEY_port_is_open);
-                if (value)
-                {
-                    is_open = *value;
-                }
-            }
+        emit portTested(ip_protocol, is_open);
+    };
 
-            emit portTested(is_open);
-        });
+    for (auto const* ip_protocol : { "ipv4", "ipv6" })
```

Comment:
_nit_: would be better as `for (auto const ip_protocol : { "ipv4"sv, "ipv6"sv }`

Reply:
No longer relevant after updating according to Mike's comments in a5aa955.